types/babel-core/index.d.ts

```diff
@@ -110,38 +156,16 @@ export interface TransformOptions {
     /** Set `file` on returned source map. */
     sourceMapTarget?: string;
 
-    /** Set `sources[0]` on returned source map. */
-    sourceFileName?: string;
-
     /** The root from which all sources are relative. */
     sourceRoot?: string;
 
-    /** Specify whether or not to use `.babelrc` and `.babelignore` files. Default: `true`. */
-    babelrc?: boolean;
-
-    /** Attach a comment before all non-user injected code. */
-    auxiliaryCommentBefore?: string;
-
-    /** Attach a comment after all non-user injected code. */
-    auxiliaryCommentAfter?: string;
-
-    /**
-     * Specify a custom callback to generate a module id with. Called as `getModuleId(moduleName)`.
-     * If falsy value is returned then the generated module id is used.
-     */
-    getModuleId?(moduleName: string): string;
-
-    /** Optional prefix for the AMD module formatter that will be prepend to the filename on module definitions. */
-    moduleRoot?: string;
+    /** Indicate the mode the code should be parsed in. Can be either “script” or “module”. Default: "module" */
+    sourceType?: "script" | "module";
 
-    /**
-     * If truthy, insert an explicit id for modules. By default, all modules are anonymous.
-     * (Not available for `common` modules).
+    /** An optional callback that can be used to wrap visitor methods.
+     *  NOTE: This is useful for things like introspection, and not really needed for implementing anything.
      */
-    moduleIds?: boolean;
-
-    /** Specify a custom name for module ids. */
-    moduleId?: string;
+    wrapPluginVisitorMethod?(pluginAlias: string, visitorType: string, callback: (path: any, state: any) => void): (path: any, state: any) => void ;
```

Comment:
> trying to get wrapPluginVisitorMethod to have better specified types – was going to ask about that actually. The callback function it takes and the function it returns can take any NodePath variant. 

That's why the parent type should be used - `NodePath` ([same as](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-3.html#generic-parameter-defaults) `NodePath<Node>`). All the variants are assignable to it, and there is no possibility to infer the specific subtype from anywhere, so the user has to use type guards or assertions anyway to work with specific subtypes. Which is unlikely however because the `wrapPluginVisitorMethod` option is intended for simpler use cases, see the original PR https://github.com/babel/babel/pull/3659.

Reply:
Since we're on it, `visitorType` can be typed as `'enter' | 'exit'`, can't it?

Reply:
Updated 👍